crates/nu-cli/src/commands/history/history_.rs

```diff
@@ -72,22 +72,23 @@ impl Command for History {
             } else {
                 let history_reader: Option<Box<dyn ReedlineHistory>> = match history.file_format {
                     HistoryFileFormat::Sqlite => {
-                        SqliteBackedHistory::with_file(history_path, None, None)
+                        SqliteBackedHistory::with_file(history_path.clone(), None, None)
```

Comment:
This `clone()` and the one below it are unfortunate, but `SqliteBackedHistory` and `FileBackedHistory`'s `with_file` methods both need `PathBuf`s rather than `impl AsRef<Path>` or something like that, and the `FileNotFound` error now also needs the path.